controllers/gatekeeper_controller.go

```diff
@@ -236,14 +237,30 @@ func (r *GatekeeperReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 
 	if err := r.handleCPSController(ctx, gatekeeper); err != nil {
-		if errors.Is(err, errCrdNotReady) {
+		if errors.Is(err, errCpsCrdNotReady) {
 			// ConstraintPodStatus CRD is not ready, wait for the CRD is ready
 			return ctrl.Result{RequeueAfter: time.Second * 3}, nil
 		} else {
 			return ctrl.Result{}, err
 		}
 	}
 
+	if err := r.handleConfigController(ctx); err != nil {
+		if errors.Is(err, errConfigCrdNotReady) {
+			// Config CRD is not ready, wait for the CRD is ready
+			return ctrl.Result{RequeueAfter: time.Second * 3}, nil
+		} else {
+			return ctrl.Result{}, err
+		}
+	}
+
+	err = r.initConfig(ctx)
+	if err != nil {
+		r.Log.Error(err, "Fail to initiate a default config resource, requeue after 3 sec")
+
+		return ctrl.Result{RequeueAfter: 3 * time.Second}, nil
```

Comment:
I'd rather rely on the controller-runtime exponential backoff here instead
```suggestion
		return ctrl.Result{}, err
```